File: src/main/java/com/market/MSA/responses/user/AuthenticationResponse.java
package com.market.MSA.responses.user;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class AuthenticationResponse {
  @JsonProperty("access_token")
  String accessToken;

  @JsonProperty("refresh_token")
  String refreshToken;

  @JsonProperty("token_type")
  final String tokenType = "Bearer";

  @JsonProperty("expires_in")
  Long expiresIn;

  boolean authenticated;
}

File: lib/feature/data/model/response/authentication_response.dart
class AuthenticationResponse {
  final String accessToken;
  final String refreshToken;
  final String tokenType;
  final int? expiresIn;
  final bool authenticated;

  AuthenticationResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.tokenType,
    required this.expiresIn,
    required this.authenticated,
  });

  factory AuthenticationResponse.fromJson(Map<String, dynamic> json) {
    return AuthenticationResponse(
      accessToken: json['access_token'] ?? '',
      refreshToken: json['refresh_token'] ?? '',
      tokenType: json['token_type'] ?? 'Bearer',
      expiresIn: json['expires_in'],
      authenticated: json['authenticated'] ?? false,
    );
  }
}
